kymatio/scattering1d/filter_bank.py

```diff
@@ -699,10 +703,16 @@ def scattering_filter_factory(J_support, J_scattering, Q, T, r_psi=math.sqrt(0.5
     # can only compute them with N=2**J_support
     for (n1, j1) in enumerate(j1s):
         N = 2**J_support
-        psi1_f.append({0: morlet_1d(
-            N, xi1[n1], sigma1[n1], normalize=normalize,
-            P_max=P_max, eps=eps)})
-
+        this_psi1_f = morlet_1d(N, xi1[n1], sigma1[n1], normalize=normalize,
+            P_max=P_max, eps=eps)
+        if spinned:
+            this_psi1_conj_f = this_psi1_f.copy()
+            this_psi1_conj_f[1:] = np.flip(this_psi1_conj_f[1:])
+            psi1_spin_f.append({0: this_psi1_conj_f, "spin": -1})
+            psi1_f.append({0: this_psi1_f, "spin": 1})
```

Comment:
i'm in favour of removing `"spin"` since it can be deduced from negative `xi` and appending `psi1_f` regardless of `spinned`

Reply:
yes @rastegah good idea. less redundancy